templates/Civi/Angular/Page/Main.tpl

```diff
@@ -1,5 +1,11 @@
+{if $route}
+  <script type="text/javascript">
+    location.hash = '{$route}';
```

Comment:
XSS vulnerability. Maybe use the Smarty `json` modifier ( https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Smarty/plugins/modifier.json.php )?